### PR TITLE
crude implementation

### PR DESCRIPTION
**Pull request by clatko-globe**
simple POC that has much room for optimization.

### CODE REVIEW

File: microcosm_pubsub/dispatcher.py [hunk 1]
```diff
@@ -36,19 +39,28 @@ def __init__(self, graph):
...
         self.send_batch_metrics = graph.pubsub_send_batch_metrics
         self.max_processing_attempts = graph.config.sqs_message_dispatcher.message_max_processing_attempts
+        self.loop = asyncio.get_event_loop()
 
     def handle_batch(self, bound_handlers) -> List[MessageHandlingResult]:
         """
         Send a batch of messages to a function.
 
         """
+        executor = concurrent.futures.ThreadPoolExecutor(5)
```
Comment (clatko-globe) [inline microcosm_pubsub/dispatcher.py]: we should add this to __init__ so we are not recreating this over and over. also the magic number should come out.

File: microcosm_pubsub/dispatcher.py [hunk 2]
```diff
@@ -36,19 +39,28 @@ def __init__(self, graph):
...
-        instances = [
-            self.handle_message(message, bound_handlers)
+        instances = []
+
+        threads = [
+            self.loop.run_in_executor(executor, self.handle_message, message, bound_handlers)
             for message in self.sqs_consumer.consume()
         ]
 
+        finished, _ = self.loop.run_until_complete(asyncio.wait(threads, return_when=asyncio.ALL_COMPLETED))
```
Comment (clatko-globe) [inline microcosm_pubsub/dispatcher.py]: the most inefficient way to wait for everything to complete.

**Comment by clatko-globe** [comment]
i went for the least intrusive way to add concurrency to the daemon, there are many other ways of doing this and many of them involve colored functions - https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/

File: microcosm_pubsub/dispatcher.py (hunk 2 above)
Comment (GabeLambda) [reply microcosm_pubsub/dispatcher.py]: Mind expounding what this means a bit?

Comment (clatko-globe) [reply microcosm_pubsub/dispatcher.py]: i'm using asyncio.ALL_COMPLETED so the arrays returned only contain finished tasks. Another way of doing this is to run within a while true to check on status of tasks and process the ones that finish as they finish.

**Pull request closed without merging**